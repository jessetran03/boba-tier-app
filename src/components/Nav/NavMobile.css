.mobile-nav-links {
  position: fixed;
  margin: 0;
  padding: 0;
  width: 100%;
  top: 84px;
  left: 0px;
  background-color: rgb(47, 173, 159, 0.97);
}

.mobile-nav-links p {
  padding: 20px 0;
  font-size: 30px;
  font-weight: 600;
  transition: color 0.2s ease-out, background-color 0.2s ease-in-out;
}

.mobile-nav-links p:hover {
  background-color: rgb(47, 173, 159, 0.97);
  color: white;
}

.mobile-nav-links .active p {
  color: white;
  background-color: rgb(83, 83, 83);
  font-weight: 600;
}

.mobile-nav-links .active p:hover {
  background-color: rgb(83, 83, 83);
}

.mobile-nav-links hr {
  margin: 0 100px;
  border: 1px solid rgb(216, 216, 216);
}