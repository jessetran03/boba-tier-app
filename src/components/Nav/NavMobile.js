import React, { Component } from 'react'
import { NavLink, Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
// import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
// import config from '../config'
import './NavMobile.css'

export default class NavMobile extends Component {

  componentDidMount() {
    document.addEventListener('mousedown', this.handleClick, false);
  }

  componentWillUnmount() {
    document.removeEventListener('mousedown', this.handleClick, false);
  }

  handleClick = (e) => {
    if (this.node.contains(e.target)) {
      return;
    }
    this.props.onHandleNav();
  }

  handleNavClick = () => {
    this.props.onHandleNav();
  }

  handleLogoutClick = () => {
    this.props.onHandleLogout()
    this.props.onHandleNav()
  }

  renderLoggedIn() {
    return (
      <>
          <NavLink exact to='/' onClick={this.handleNavClick} activeClassName='active'>
            <p>Home</p>
          </NavLink>
          <hr />
          <NavLink to='/shops' onClick={this.handleNavClick} activeClassName='active'>
            <p>Shop List</p>
          </NavLink>
          <hr />
          <NavLink to='/rankings' onClick={this.handleNavClick} activeClassName='active'>
            <p>Top 10 Drinks</p>
          </NavLink>
          <hr />
          <NavLink to='/lists' onClick={this.handleNavClick} activeClassName='active'>
            <p>My Ratings</p>
          </NavLink>
          <hr />
          <Link onClick={this.handleLogoutClick} to='/' >
            <p>Log Out</p>
          </Link>
      </>
    )
  }

  renderLoggedOut() {
    return (
      <>
        <NavLink exact to='/' onClick={this.handleNavClick} activeClassName='active' >
          <p>Home</p>
        </NavLink>
        <hr />
        <NavLink to='/shops' onClick={this.handleNavClick} activeClassName='active'>
          <p>Shop List</p>
        </NavLink>
        <hr />
        <NavLink to='/rankings' onClick={this.handleNavClick} activeClassName='active'>
          <p>Top 10 Drinks</p>
        </NavLink>
        <hr />
        <NavLink to='/login' onClick={this.handleNavClick} activeClassName='active'>
          <p>Log In</p>
        </NavLink>
        <hr />
        <NavLink to='/register' onClick={this.handleNavClick} activeClassName='active'>
          <p>Sign Up</p>
        </NavLink>
      </>
    )
  }

  render() {
    return (
      <section ref={node => this.node = node} className='mobile-nav-links mobile-nav'>
        {TokenService.hasAuthToken()
          ? this.renderLoggedIn()
          : this.renderLoggedOut()
        }
      </section>
    )
  }
}