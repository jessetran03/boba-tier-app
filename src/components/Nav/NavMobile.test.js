import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import NavMobile from './NavMobile'
import renderer from 'react-test-renderer'

it('renders NavMobile without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <NavMobile />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('render NavMobile without data', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <NavMobile />
    </BrowserRouter>
  )
    .toJSON();
  expect(wrapper).toMatchSnapshot()
})