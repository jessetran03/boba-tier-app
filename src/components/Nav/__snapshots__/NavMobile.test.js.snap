// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render NavMobile without data 1`] = `
<section
  className="mobile-nav-links mobile-nav"
>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <p>
      Home
    </p>
  </a>
  <hr />
  <a
    aria-current={null}
    href="/shops"
    onClick={[Function]}
  >
    <p>
      Shop List
    </p>
  </a>
  <hr />
  <a
    aria-current={null}
    href="/rankings"
    onClick={[Function]}
  >
    <p>
      Top 10 Drinks
    </p>
  </a>
  <hr />
  <a
    aria-current={null}
    href="/login"
    onClick={[Function]}
  >
    <p>
      Log In
    </p>
  </a>
  <hr />
  <a
    aria-current={null}
    href="/register"
    onClick={[Function]}
  >
    <p>
      Sign Up
    </p>
  </a>
</section>
`;
